packages/shr-fhir-export/lib/extensions.js

```diff
@@ -120,6 +120,7 @@ class ExtensionExporter {
 
     // If 'type' is not defined, that means we can't represent this as a simple extension
     if (type === undefined) {
+      this.zeroOutValueXElement(extension);
```

Comment:
NOTE: This was moved up higher in the function body because some extensions get looked up and used before they are done being processed themselves (this is a circular dependency thing, @kjmahalingam!).  There was code that was looking at the `value[x]` element to see if it was zeroed out and when the code to zero it out was further down the function body, the `value[x]` hadn't yet been zeroed out when other code looked at it -- even though it was eventually going to be zeroed out.  So I moved it up to ensure it is zeroed out earlier than later.

Reply:
I'm re-reading this comment now.  Super confusing.  Sorry.  Let me know if you have any questions.